src/main/scala/thurloe/dataaccess/HttpSendGridDAO.scala

```diff
@@ -0,0 +1,81 @@
+package thurloe.dataaccess
+
+import com.sendgrid.SendGrid.Response
+import com.sendgrid._
+import com.typesafe.config.ConfigFactory
+import thurloe.database.ThurloeDatabaseConnector
+import thurloe.service.Notification
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+/**
+ * Created by mbemis on 6/16/16.
+ */
+class HttpSendGridDAO {
+
+  val configFile = ConfigFactory.load()
+  val sendGridConfig = configFile.getConfig("sendgrid")
+  val apiKey = sendGridConfig.getString("apiKey")
+  val substitutionChar = sendGridConfig.getString("substitutionChar")
+  val fromAddress = sendGridConfig.getString("defaultFromAddress")
+
+  val dataAccess = ThurloeDatabaseConnector
+
+  def sendNotification(notification: Notification): Future[Response] = {
+    lookupPreferredEmail(notification.userId) flatMap { preferredEmail =>
+      val email = createEmail(preferredEmail, notification.notificationId, notification.substitutions)
+      sendEmail(email)
+    }
+  }
+
+  /*
+    Note: email.setSubject and email.setText must be set even if their values
+    aren't used. Supposedly this will be fixed in a future version of SendGrid
+   */
+  def createEmail(toAddress: String, notificationId: String, substitutions: Map[String, String] = Map.empty): SendGrid.Email = {
+    val email = new SendGrid.Email()
+
+    email.addTo(toAddress)
+    email.setFrom(fromAddress)
+    email.setTemplateId(notificationId)
+    email.setSubject(" ")
+    email.setText(" ")
+    addSubstitutions(email, substitutions)
+  }
+
+  def sendEmail(email: SendGrid.Email): Future[Response] = {
+    val sendGrid = new SendGrid(apiKey)
+
+    Future {
+      val response = sendGrid.send(email)
+      if(response.getStatus) response
+      else throw new NotificationException
+    }
+  }
+
+  def lookupPreferredEmail(userId: String): Future[String] = {
+    for {
+      contactEmail <- dataAccess.lookup(userId, "contactEmail")
+      email <- dataAccess.lookup(userId, "email")
+    } yield if(contactEmail.keyValuePair.value.isEmpty) email.keyValuePair.value else contactEmail.keyValuePair.value
```

Comment:
one of my goals with this endpoint was to make it unaware of who was consuming the endpoint. having a fallback to the google account would mean that we can't use this same code for "You have been added to X workspace" or "Your submission has completed" notifications because the user who should be notified isn't necessarily the same user that prompted the notification to be sent